### ISSUE TITLE
Add a shortcut to jump to any model instance

### ISSUE THREAD

**Issue by knyghty** [knyghty/django-admin-keyboard-shortcuts#5]
I would guess we can somehow hook into `search_fields` to to have a shortcut that:

When `g` is pressed followed by `c` (for change, could be something else?), a dialog should pop up.

This should have a selectable list of models reused from the `g` `l` shortcut.

When a model is selected, a new input should appear in which you can type in a string that will run a partial match on that model, probably using the admin's `search_fields` functionality (somehow). Edit: thinking on this further, `autocomplete_fields` probably makes more sense as it's more functionally similar.

Then you click choose one of the results and go directly to the change list page for that instance.

**Comment by khanxmetu** [comment]
> thinking on this further, autocomplete_fields probably makes more sense as it's more functionally similar.  

I agree. Autocomplete view is similar to this feature implementation wise. It internally calls `ModelAdmin().get_search_results()` which performs the search and returns the resulting queryset.  

I was also wondering if the api/json view of this feature could somehow be merged into ChangeList view and ChangeListSearchForm. For example:
```
GET /admin/demo/artist/?q=bob
Accept: application/json
```
Should return the search results as json.
But the functionality of ChangeList seems way too broad with filtering, ordering, what fields to display, etc, which may be quite unnecessary if we are going for a simple quick search (in command pallete for example).

**Comment by ChathuraGH** [comment]
> I would guess we can somehow hook into search_fields to to have a shortcut that:
> 

Do we need?  For text fields, we can just get models the have search enabled suggest these models to select, and build the required url with given model and search query text. 



> When g is pressed followed by c (for change, could be something else?), a dialog should pop up.
> 

We can just use `Ctrl+k`. And use the same command palette. And suggest models with partial matches for model names.

 

> This should have a selectable list of models reused from the g l shortcut.
> 
> When a model is selected, a new input should appear in which you can type in a string that will run a partial match on that model, probably using the admin's search_fields functionality (somehow). Edit: thinking on this further, autocomplete_fields probably makes more sense as it's more functionally similar.
> 

We can use the same input field and use a separation character. (example : "|" ). No need to type these characters. It's auto filled. 



> Then you click choose one of the results and go directly to the change list page for that instance.


We can just point the user to to search result pages.





Please give your feedback on this.

**Comment by merwok** [comment]
> run a partial match on that model, probably using the admin's search_fields functionality (somehow). Edit: thinking on this further, autocomplete_fields probably makes more sense 

You can check how autocomplete_fields work! It’s already using a json endpoint that relies on search_fields (configured in the admin of the model you’re completing for) to find results. So you would definitely reuse that.

**Comment by ChathuraGH** [comment]
I guess then, there are two options.

1. Redirect user to Search result page, - if user didn't pick a specific auto suggested object 
2. Redirect user to specific one object edit page, - if user selected/pick a specific one object from auto suggestion

Like in Google search, while you can pick a suggestion, you can also reach specific search result page. 

I'm just thinking in design area.

**Comment by ChathuraGH** [comment]
Humm. I just tinkerd some code and it works.

**autocomplete_fields work!**


Thanks for pointing it out. I was just thinking about search result pages and backend changes.

With a little bit of JS, it works.

To note here, hard part was the, figuring out url parameter names. Strange right! `q=query_string` doesn't work.

Still on my local test project. 


🚀

**Comment by merwok** [comment]
> ```
>	# autocomplete_fields =['title','content']
>
> #autocomplete_fields are not required .
> ```

Still a misunderstanding here: autocomplete_fields would be for example for the `"author"` field inside the PostAdmin, and it would rely on `search_fields` in the UserAdmin.

**Comment by ChathuraGH** [comment]
Ok. I understand it's for foringnkey and mtm fields.

**Comment by ChathuraGH** [comment]
knyghty also pointed out about command complexity with `?` and `|`. It's possible to eliminate that.(with using js to remember last selected command/model). 

But I like to point out that even github uses a separator character. (:).

![Image](https://github.com/user-attachments/assets/80f87ebf-5c7a-4bbb-b90e-b0fb17d4d7cf)

I think it should be decided by @knyghty and other community members.(or some more refined and simple structure). 

Thank you.

**Comment by ChathuraGH** [comment]
> I would guess we can somehow hook into `search_fields` to to have a shortcut that:


Quick question here,

Let's say user press a key combination in an invalid location(page).

Like, (`alt+s`) to save current editing object is the expected behaviour.
But he presses it on home page. 

What should the design path take? 

- capture the event, notify the user about it's usage 
- don't capture the event

**Comment by ChathuraGH** [comment]
Extra considerations update.



- [x] Highlight query text in result list(phrase matchings)
- [ ] delay the query execution for reduced query frequency (or do we need that?, query for each character change)
- [ ] Highlight the actual query string inside the input (use different colours for model name and query string)
